src/main.rs

```diff
@@ -46,39 +49,45 @@ async fn main() -> Result<(), Box<dyn Error>> {
                 .filter(|expiry| expiry > &&chrono::Utc::now())
                 .is_some()
             {
-                println!(
+                debug!(
                     "Found device {}, ignoring because of backoff",
                     display_device(device.as_ref())
                 );
                 continue;
             }
-            println!(
+            info!(
                 "Found device {}, connecting",
                 display_device(device.as_ref())
             );
             if let Err(e) = device.connect(&session).await {
-                eprintln!("Failed to connect, skipping: {:?}", e);
+                error!("Failed to connect, skipping: {:?}", e);
                 continue;
             }
 
-            println!("Getting data");
+            info!("Getting data from {}", display_device(device.as_ref()));
             match device.get_data(&session).await {
                 Ok(data) => {
-                    println!(
-                        "Fetched {} records, last 5: {:?}",
-                        data.len(),
-                        data.iter().rev().take(5).collect::<Vec<_>>()
-                    );
-                    backoff_table.insert(
-                        device.get_device_info().id.clone(),
-                        chrono::Utc::now()
-                            .checked_add_signed(chrono::Duration::minutes(5))
-                            .unwrap(),
+                    info!("Fetched {} records", data.len());
+                    debug!("Last 5 records loaded",);
+                    data.iter()
+                        .rev()
+                        .take(5)
+                        .for_each(|record| debug!("{:?}", record));
+
+                    let backoff_expiry = chrono::Utc::now()
+                        .checked_add_signed(chrono::Duration::minutes(5))
+                        .unwrap();
+                    debug!(
```

Comment:
Nit: why not make it info? It could be useful (outside of development) to know how long the device will be ignored.
```suggestion
                    info!(
```